lib/DebugTrackWastedRenders.lua

```diff
@@ -0,0 +1,115 @@
+local DebugTrackWastedRenders = {}
+
+-- Tracks total number of renders by type of component
+local totalRenders = 0
+local wastedRendersByType = {}
```

Comment:
To make this less error-prone, let's make this have a default of 0 with metatables:

```lua
local wastedRendersByType = setmetatable({}, {
    __index = function(self, key)
        self[key] = 0
        return 0
    end,
})
```

and the same for `validRendersByType`.

Reply:
The other thing we can do is just default initialize each component type manually, which would make it so we can still iterate over the table.